internal/app/commands/project/deploy.go

```diff
@@ -32,22 +32,33 @@ var Deploy = cli.Command{
 		force := common.IsForceMode(c)
 		continuous := c.Bool("continuous")
 		if projectData := ensureProjectDataExists(c, ".", "A sandbox is required to deploy the project, do you want to create one?", true); projectData != nil {
+			sandboxExists := sandbox.Exists(projectData.Sandbox)
 			tasks := []string{"deploy"}
 			if continuous {
 				tasks = append(tasks, "--continuous")
 				// ask to run sandbox in detached mode before gradle deploy because it has continuous flag
-				askToRunSandbox(c, projectData)
-				fmt.Fprintln(os.Stderr, "")
+				if sandboxExists {
+					askToRunSandbox(c, projectData)
+					fmt.Fprintln(os.Stderr, "")
+				}
 			}
 
-			runGradleTask(projectData, fmt.Sprintf("Deploying to sandbox '%s'...", projectData.Sandbox), tasks...)
+			var deployMessage string
+			if sandboxExists {
+				deployMessage = fmt.Sprintf("Deploying to sandbox '%s'...", projectData.Sandbox)
+			} else {
+				deployMessage = "No sandbox found, deploying without a sandbox..."
```

Comment:
Deploying without a sandbox. What is suppose to happen in that case?

Reply:
It runs **gradle deploy** without providing xp home path and xp java, so it will use system java and will not copy jar to xp home.